From f503ecfd035951b6d6aac51c182fef98efdd6447 Mon Sep 17 00:00:00 2001
From: Luca Fancellu <luca.fancellu@arm.com>
Date: Mon, 3 Apr 2023 08:45:27 +0100
Subject: [PATCH] net: ip: Kconfig for TCP packet allocation timeout

TCP packet allocation timeout is currently 100ms, but there are cases
where it is not enough and as a side effect, the kernel internals are
printing some errors on the log before retrying again, create a
Kconfig parameter to be able to tune this value.

Issue-Id: SCM-5838
Signed-off-by: Luca Fancellu <luca.fancellu@arm.com>
Change-Id: Id490682a8de2b9bb94f310fee9c96e45b135e00d
Upstream-Status: Inappropriate
---
 subsys/net/ip/Kconfig       | 10 ++++++++++
 subsys/net/ip/tcp_private.h |  2 +-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/subsys/net/ip/Kconfig b/subsys/net/ip/Kconfig
index 40a25c3a4f57..839d5b96576c 100644
--- a/subsys/net/ip/Kconfig
+++ b/subsys/net/ip/Kconfig
@@ -494,6 +494,16 @@ config NET_TCP_RECV_QUEUE_TIMEOUT
 	  SEQ 2. But if we receive SEQs 5,4,3,7 then the SEQ 7 is discarded
 	  because the list would not be sequential as number 6 is be missing.
 
+config NET_TCP_PKT_ALLOC_TIMEOUT
+	int "How long to wait for a TCP packet allocation (in ms)"
+	depends on NET_TCP
+	default 100
+	help
+	  The TCP network stack allocates packets from the buffers and the
+	  allocation can take some time depending on the situation.
+	  This value means how long should the stack wait for the packet to be
+	  allocated, before returning an internal error and try again.
+
 config NET_TCP_WORKQ_STACK_SIZE
 	int "TCP work queue thread stack size"
 	default 1024
diff --git a/subsys/net/ip/tcp_private.h b/subsys/net/ip/tcp_private.h
index 952e13939997..2fcd38a70645 100644
--- a/subsys/net/ip/tcp_private.h
+++ b/subsys/net/ip/tcp_private.h
@@ -45,7 +45,7 @@
 #define tcp_free(_ptr) k_free(_ptr)
 #endif
 
-#define TCP_PKT_ALLOC_TIMEOUT K_MSEC(100)
+#define TCP_PKT_ALLOC_TIMEOUT K_MSEC(CONFIG_NET_TCP_PKT_ALLOC_TIMEOUT)
 
 #if defined(CONFIG_NET_TEST_PROTOCOL)
 #define tcp_pkt_clone(_pkt) tp_pkt_clone(_pkt, tp_basename(__FILE__), __LINE__)
-- 
2.34.1

